cypress/e2e/CMS/$Event/index.js

```diff
@@ -0,0 +1,177 @@
+And('user selects {int} Promotional Image from the list of images', (num) => {
+    cy.get('span:contains("Promotional Image")').parent().first().as('imageUpload').click()
+    cy.get('input[name="field_image_promotional_entity_browser_entity_browser"]').click({ force: true })
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find("input[id^='edit-entity-browser-select-media']").eq(num - 1).check()
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find("input[id='edit-submit'][value='Select image']").click({ force: true })
+});
+
+And('user remembers the source of selected lead image for further verification', () => {
+    cy.get('details img').then($el => {
+        imageSrc = $el[0].getAttribute('src')
+    });
+});
+
+And('the Event Series title matches selected event series', () => {
+    let textSrc;
+    let expectedText;
+    cy.get('#cgvBody h2').then($el => {
+        textSrc = $el[0].innerText;
+        expectedText = textSrc.trim();
+        cy.get('h2.event-series').should('include.text', expectedText);
+    });
+});
+
+And('the event date is displaying todays date', () => {
+    const months = [
+        'January',
+        'February',
+        'March',
+        'April',
+        'May',
+        'June',
+        'July',
+        'August',
+        'September',
+        'October',
+        'November',
+        'December'
+    ];
+    const date = new Date();
+    const year = date.getFullYear();
+    const month = months[date.getMonth()];
+    const day = date.getDate();
+    const expectedDate = `${month} ${day}, ${year}`;
+    cy.get('p.event-date-time').should('include.text', expectedDate);
+});
+
+And('the event start time displays in the following format {string}', (num) => {
+    cy.get('p.event-date-time').invoke('text').then((time) => {
+        const startAndEnd = time.split('|');
+        const startAndEndtime = startAndEnd[1].split('–');
+        const regex = new RegExp(num);
+        expect(startAndEndtime[0]).to.match(regex);
+    });
+});
+
+And('the event end time displays {string}', (num) => {
+    cy.get('p.event-date-time').should('include.text', num);
+});
+
+And('the Venue location field displays {string}', (locVenue) => {
+    cy.get('span.event-location-venue').should('have.text', locVenue);
+});
+
+And('the room location field displays {string}', (roomNum) => {
+    cy.get('span.event-location-room').should('have.text', roomNum);
+});
+
+And('the city-state location field displays {string}', (cityState) => {
+    cy.get('span.event-location-city-state').should('have.text', cityState);
+});
+
+And('{string} is a link with href that matches {string}', (linkText, linkHref) => {
+    const regex = new RegExp(linkHref);
+    cy.get('div#cgvBody p').find('a').should('have.text', linkText);
+    cy.get('div#cgvBody p').find('a').should('have.attr', 'href').and('match', regex);
+});
+
+And('description reads {string}', (desc) => {
+    cy.get('#cgvBody > :nth-child(6)').should('be.visible')
```

Comment:
cy.get(``p:contains(${desc})``).should('be.visible')

Reply:
done